lib/decorators/validations/array-distinct.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+} from 'class-validator';
+
+/**
+ * Custom validation decorator used to check if array contains distinct values.
+ * 
+ * @param {string} property 
+ * @param {ValidationOptions?} validationOptions 
+
+ */
+export default function ArrayDistinct(
+  property: string,
+  validationOptions?: ValidationOptions,
+) {
+  return (object: any, propertyName: string): void => {
+    registerDecorator({
+      name: 'ArrayDistinct',
+      target: object.constructor,
+      propertyName,
+      constraints: [property],
+      options: validationOptions,
+      validator: {
+        validate(value: any): boolean {
+          if (Array.isArray(value)) {
+            const distinct = [
+              ...new Set(
+                value.map((v): Record<string, unknown> => v[property]),
+              ),
+            ];
+            return distinct.length === value.length;
+          }
+          return false;
+        },
+        defaultMessage(args: ValidationArguments): string {
+          return `${args.property} contains duplicate ${args.constraints[0]} values.`;
```

Comment:
May be change this to

```ts
return `${args.property} must be an array and should not contain duplicate values.`;
```

since the function returns false if the value is not an array as well.